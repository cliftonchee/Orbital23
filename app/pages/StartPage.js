import { StyleSheet, Text, View, Image, SafeAreaView } from "react-native";
import colours from "../components/Colours";
import Button from "../components/Button";

// To navigate between screens, add '{ navigation }'
// as an argument, and under onPress, refer to below
export default function StartPage({ navigation }) {
  return (
    <SafeAreaView style={styles.container}>
      <Image
        style={styles.image}
        source={require("../assets/images/polaris-constellation.png")}
      />

      {/* Title */}
      <Text style={styles.title}>P O L A R I S</Text>
      {/* Subtitle */}
      <Text style={styles.subtitle}>Connect with the stars</Text>

      {/* Customisable Button */}
      <View style={styles.button}>
        <Button
          styleOverride={styles.buttonLayout}
          // Navigation to Login page
          onPress={() => navigation.navigate("Login")}
          title="Get started"
        ></Button>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    // Background
    flex: 1,
    alignItems: "center",
    backgroundColor: colours.primary,
  },
  image: {
    flex: 3,
    justifyContent: "center",
    maxWidth: 960,
  },
  title: {
    flex: 0.5,
    alignItems: "center",
    fontSize: 50,
    color: colours.white,
    fontFamily: "LatoBold",
  },
  subtitle: {
    flex: 0.4,
    fontSize: 24,
    color: colours.subtitle,
    fontFamily: "LatoRegular",
  },
  buttonLayout: {
    // Added on top of original button
    borderRadius: 30,
    width: 350,
    height: 100,
  },
  button: {
    flex: 1,
  },
});

// import React, { useRef, useEffect } from 'react';
// import UnityView from '@azesmway/react-native-unity';
// import { View } from 'react-native';

// const AR = () => {
//   const unityRef = useRef(null);

//   useEffect(() => {
//     if (unityRef.current) {
//       const message = {
//         gameObject: 'gameObject',
//         methodName: 'methodName',
//         message: 'message',
//       };
//       unityRef.current.postMessage(message.gameObject, message.methodName, message.message);
//     }
//   }, []);

//   return (
//     <View style={{ flex: 1 }}>
//       <UnityView
//         ref={unityRef}
//         style={{ flex: 1 }}
//         onUnityMessage={(result) => {
//           console.log('onUnityMessage', result.nativeEvent.message);
//         }}
//       />
//     </View>
//   );
// };

// export default AR;
