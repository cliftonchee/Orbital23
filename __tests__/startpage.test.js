// import React from 'react';
// import { NavigationContainer } from '@react-navigation/native';
// import { render, screen, fireEvent } from '@testing-library/react-native';
import renderer from "react-test-renderer";

import StartPage from "../app/pages/StartPage";
// import TestAppNavigator from '../app/navigation/Navigation';

// Run 'jest' OR 'yarn test' in terminal to test

// Snapshot Testing: If there is a need to update snapshot,
// use 'jest --UpdateSnapshot' but use with caution
// as error may be due to our fault,
// can refer to this: https://jestjs.io/docs/snapshot-testing 

it('Start page renders correctly', () => {
  const tree = renderer
    .create(<StartPage />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// Testing React Navigation from Start Page (Unsure how to implement)

// describe('Testing React Navigation', () => {
//   test('screen contains a button linking to the login page', async () => {
//     const component = (
//       <NavigationContainer>
//         <TestAppNavigator />
//       </NavigationContainer>
//     );

//     render(component);
//     const button = await screen.findByText('Go to notifications');

//     expect(button).toBeOnTheScreen();
//   });

//   test('clicking on the button takes you to the notifications screen', async () => {
//     const component = (
//       <NavigationContainer>
//         <TestAppNavigator />
//       </NavigationContainer>
//     );

//     render(component);
//     const title = screen.queryByText('P O L A R I S');
//     const subtitle = screen.queryByText('Connect with the stars');
//     const button = await screen.findByText('Get started');

//     expect(title).toBeOnTheScreen();
//     expect(subtitle).toBeOnTheScreen();

//     fireEvent(button, 'press');
//     const newScreen = await screen.findByText('Forgot Password');

//     expect(newScreen).toBeOnTheScreen();
//   });
// });