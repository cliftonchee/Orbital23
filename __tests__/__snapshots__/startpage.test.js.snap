// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> Start page renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#201C33",
      "flex": 1,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../app/assets/images/polaris-constellation.png",
      }
    }
    style={
      {
        "flex": 3,
        "justifyContent": "center",
        "maxWidth": 960,
      }
    }
  />
  <Text
    style={
      {
        "alignItems": "center",
        "color": "#fff",
        "flex": 0.5,
        "fontFamily": "LatoBold",
        "fontSize": 50,
      }
    }
  >
    P O L A R I S
  </Text>
  <Text
    style={
      {
        "color": "#626070",
        "flex": 0.4,
        "fontFamily": "LatoRegular",
        "fontSize": 24,
      }
    }
  >
    Connect with the stars
  </Text>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#100428",
            "borderRadius": 30,
            "height": 100,
            "justifyContent": "center",
            "margin": 5,
            "opacity": 1,
            "width": 350,
          }
        }
        testID="button"
      >
        <Text
          style={
            [
              {
                "color": "#fff",
                "fontFamily": "LatoBold",
                "fontSize": 20,
              },
              undefined,
            ]
          }
        >
          Get started
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
